### PR TITLE
Add CodeQL security and quality scan

### LINKED ISSUE #1256: Add a security scanning tool to the build pipeline (e.g. CodeQL)
**Is your feature request related to a problem? Please describe.**

As recommended by the Zowe TSC, squads will choose and integrate a solution for security scanning

**Describe the solution you'd like**

Suggested tool: CodeQL
Frequency: Once a week? and before every release

**Describe alternatives you've considered**

**Additional context**

This tool is already being used by the Zowe APIML Squad as part of their pipelines

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Resolves #1256 by adding automated CodeQL scans on PRs and addressing high severity issues that it found

## Release Notes

Milestone:

Changelog:

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 1]
```diff
@@ -329,21 +329,33 @@ export async function initializeZoweFolder(): Promise<void> {
...
-        if (content && content?.overrides) {
-            if (content?.overrides?.CredentialManager === globals.PROFILE_SECURITY) {
+    try {
+        fd = fs.openSync(settingsFile, "r+");
+        fileContent = fs.readFileSync(fd, "utf-8");
+    } catch {
+        // If reading the file failed because it does not exist, then create it
+        // This should never fail, unless file system is read-only or the file
+        // was created by another process after first openSync call
+        fd = fs.openSync(settingsFile, "wx");
```
Comment (traeok) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:342]: line 342: Seems like if `openSync` fails here, it won't be caught. Do we have to wrap this in another `try/catch`?

It seems highly unlikely that this situation would occur, but we might still want to catch an exception here if one pops up.

**Review by traeok — APPROVED** [review]
Left a question about catching 2nd `openSync` call

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts (hunk 1 above)
Comment (t1m0thyj) [reply packages/zowe-explorer/src/utils/ProfilesUtils.ts:342]: Good catch, I'm not sure in the `writeOverridesFile` method whether we want to catch all possible errors, or catch only the ones we need to handle and leave it to the caller to handle other errors.

If catching all possible errors is the desired behavior, then we should also catch and handle errors thrown by the `JSON.parse` call below.

**Review by rudyflores — APPROVED** [review]
Thanks for these changes Timothy! seems to be working as intended!

**Review by zFernand0 — DISMISSED** [review]

**Review by zFernand0 — APPROVED** [review]

**Review by traeok — APPROVED** [review]

**Pull request merged**